avro/src/test/scala/magnolify/avro/AvroTypeSuite.scala

```diff
@@ -190,7 +197,29 @@ class AvroTypeSuite extends MagnolifySuite {
   test[MapPrimitive]
   test[MapNested]
 
-  test[Logical]
+  {
+    // generate unsigned-int duration values
+    implicit val arbAvroDuration: Arbitrary[AvroDuration] = Arbitrary {
+      for {
+        months <- Gen.chooseNum(0L, 0xffffffffL)
+        days <- Gen.chooseNum(0L, 0xffffffffL)
+        millis <- Gen.chooseNum(0L, 0xffffffffL)
+      } yield AvroDuration(months, days, millis)
+    }
+
+    implicit val afAvroDuration: AvroField[AvroDuration] = AvroField.from[(Long, Long, Long)] {
+      case (months, days, millis) => AvroDuration(months, days, millis)
+    }(d => (d.months, d.days, d.millis))(AvroField.afDuration)
```

Comment:
We either need explain how to use `AvroField.afDuration` in docs, or introduce `AvroDuration` in magnolify. It is concerning that it won't be useful to users

Reply:
I'm not for exposing a custom `AvroDuration` class. IHMO, the avro java lib does not expose the `duration` logical type because there does not exists a proper java data structure to handle the value as defined in the spec.

In the end `afDuration` is not different from any other type, so users should not be lost.

Reply:
What I was thinking is that it might be not obvious for a user how to discover `afDuration`

Reply:
We can include that in #612 . We'll need to create a proper doc for all modules